src/Symfony/Component/Yaml/Parser.php

```diff
@@ -964,7 +974,7 @@ private function isCurrentLineBlank(): bool
     private function isCurrentLineComment(): bool
     {
         //checking explicitly the first char of the trim is faster than loops or strpos
-        $ltrimmedLine = ltrim($this->currentLine, ' ');
+        $ltrimmedLine = ' ' === $this->currentLine[0] ? ltrim($this->currentLine, ' ') : $this->currentLine;
```

Comment:
What about rewriting the method to something like the following?

```php
private function isCurrentLineComment(): bool
{
    if ('' === $this->currentLine) {
        return false;
    }

    if ('#' === $this->currentLine[0]) {
        return true;
    }

    $ltrimmedLine = ltrim($this->currentLine, ' ');

    return '' !== $ltrimmedLine && '#' === $ltrimmedLine[0];
}
```

Reply:
Equally valid. If you prefer this expanded version I'd be happy to use it.